ios/Exponent/ExpoKit/ExpoKit.h

```diff
@@ -29,6 +29,11 @@ FOUNDATION_EXPORT NSString * const EXAppDidRegisterUserNotificationSettingsNotif
  */
 - (UIViewController *)currentViewController;
 
+/**
+ *  Set up dependencies that need to be initialized before  app delegates.
+ */
+- (void)setUpApplication:(UIApplication *)application withLaunchOptions:(nullable NSDictionary *)launchOptions;
```

Comment:
Neat solution! Clever thinking! 🎖️

One thing I would maybe change in this method is its similarity to the `application:didFinishLaunchingWithOptions:` callback. IMHO right now they're quite similar in that they both accept the same arguments and both revolve around `application` and `launchOptions`—due to this it's not easy to grasp immediately what's the difference between them and what each one is responsible for.

I can't think of a name that strikes me as perfect for this method. Only `setUpWithLaunchOptions:`, `provideLaunchOptions:`, `prepareWithLaunchOptions:` come to mind. What do you think?